src/v/hashing/tests/hash_bench.cc

```diff
@@ -11,64 +11,259 @@
-    auto o = crc.value();
-    perf_tests::do_not_optimize(o);
-    perf_tests::stop_measuring_time();
+
+// wrap an sstring so that it can define AbslHashValue
+struct abseil_wrapper_cc {
+    ss::sstring& _s;
+
+    template<typename H>
+    friend H AbslHashValue(H h, const abseil_wrapper_cc& s) {
+        return H::combine_contiguous(std::move(h), s._s.data(), s._s.size());
+    }
+};
+
+PERF_TEST(header_hash, abseil_combine_contiguous) {
+    return header_body([](ss::sstring& buffer) {
+        return absl::HashOf(abseil_wrapper_cc{buffer});
+    });
 }
 
-PERF_TEST(xx32_fn, header_hash) {
-    auto buffer = random_generators::gen_alphanum_string(step_bytes);
-    perf_tests::start_measuring_time();
-    auto o = xxhash_32(buffer.data(), buffer.size());
-    perf_tests::do_not_optimize(o);
-    perf_tests::stop_measuring_time();
+// wrap an sstring so that it can define AbslHashValue
+struct abseil_wrapper_string_view {
+    ss::sstring& _s;
+
+    template<typename H>
+    friend H AbslHashValue(H h, const abseil_wrapper_string_view& s) {
+        return H::combine(
+          std::move(h), absl::string_view(s._s.data(), s._s.size()));
+    }
+};
+
+PERF_TEST(header_hash, abseil_wrap_string_view) {
+    return header_body([](ss::sstring& buffer) {
+        return absl::HashOf(abseil_wrapper_string_view{buffer});
+    });
 }
 
-PERF_TEST(xx64_twang_fn, header_hash) {
-    auto buffer = random_generators::gen_alphanum_string(step_bytes);
+template<typename T>
+static auto std_hash_of(const T& t) {
+    return std::hash<T>{}(t);
+}
+
+PERF_TEST(header_hash, std_hash_string_view) {
+    return header_body([](ss::sstring& buffer) {
+        return std_hash_of(std::string_view(buffer.data(), buffer.size()));
+    });
+}
+
+PERF_TEST(header_hash, sstring) {
+    return header_body([](ss::sstring& buffer) { return std_hash_of(buffer); });
+}
+
+PERF_TEST(header_hash, boost_crc16_fn) {
+    return header_body([](auto& buffer) {
+        boost::crc_16_type crc;
+        crc.process_bytes(buffer.data(), buffer.size());
+        return crc.checksum();
+    });
+}
+
+PERF_TEST(header_hash, boost_ccitt16_fn) {
+    return header_body([](auto& buffer) {
+        boost::crc_ccitt_type crc;
+        crc.process_bytes(buffer.data(), buffer.size());
+        return crc.checksum();
+    });
+}
+
+PERF_TEST(header_hash, fnv32_fn) {
+    return header_body(
+      [](auto& buffer) { return fnv32_buf(buffer.data(), buffer.size()); });
+}
+
+PERF_TEST(header_hash, crc32_fn) {
+    return header_body([](auto& buffer) {
+        crc::crc32c crc;
+        crc.extend(buffer.data(), buffer.size());
+        return crc.value();
+    });
+}
+
+PERF_TEST(header_hash, xx32_fn) {
+    return header_body(
+      [](auto& buffer) { return xxhash_32(buffer.data(), buffer.size()); });
+}
+
+PERF_TEST(header_hash, xx64_twang_fn) {
+    return header_body([](auto& buffer) {
+        return twang_32from64(xxhash_64(buffer.data(), buffer.size()));
+    });
+}
+
+using model::ktp;
```

Comment:
Everything below is mostly trying various ntp hash methods, without too many surprises. Absl hashers are generally good but libc++ `string_view` hash is actaully slightly faster at these sizes.